src/site/content/en/blog/native-file-system/index.md

```diff
@@ -184,12 +185,50 @@ async function getNewFileHandle() {
 }
 ```
 
-#### Save changes to the original file
+#### Save changes to disk {: #save-to-disk }
 
 You can find all the code for saving changes to a file in my [text
 editor][text-editor] demo on [GitHub][text-editor-source]. The core file system
 interactions are in [`fs-helpers.js`][text-editor-fs-helper]. At its simpliest,
-the process looks like the code below. I'll walk through each step and explain it.
+the process looks like the code below. I'll walk through each step and explain
+it.
+
+```js
+async function writeFile(fileHandle, contents) {
+  // Create a FileSystemWritableFileStream to write to.
+  const writable = await fileHandle.createWritable();
+  // Write the contents of the file to the stream.
+  await writable.write(contents);
+  // Close the file and write the contents to disk.
+  await writable.close();
+}
+```
+
+Writing data to disk uses a [`FileSystemWritableFileStream`][fs-writablestream],
+essentially a [`WritableStream`][writable-stream]. Create the stream by calling
+`createWritable()` on the file handle object. When `createWritable()` is
+called, Chrome first checks if the user has granted write permission to the
+file. If permission to write hasn't been granted, the browser will prompt
+the user for permission. If permission isn't granted, `createWritable()`
+will throw a `DOMException`, and the app will not be able to write to the
+file. In the text editor, these `DOMException`s are handled in the
+[`saveFile()`][text-editor-app-js] method.
+
+The `write()` method takes a string, which is what we want for a text editor.
+But it can also take a [BufferSource][buffersource], or a [Blob][blob].
+Unlike most `WritableStreams`, changes are **not** written to disk until the
+stream is closed by calling `close()`.
+
```

Comment:
is it worth having some example that actually uses the fact that this is a WritableStream as well? The current example doesn't really use any stream specific bits, instead it just uses our convenience wrappers around what is provided by WritableStream. Not sure what such an example would look like (perhaps something that pipes the result of fetch() to a file?)

Other thoughts on this area is that perhaps we should call out that this is effectively a seekable writable stream? I.e. it's a normal writable stream, but you can also seek in it.

Reply:
Added an example using fetch and piping the response, and a sentence about `seek` and `truncate`.